### ISSUE TITLE
OpenAI Realtime API Beta is not compatible with Bun runtime

### ISSUE THREAD

**Issue by K-Mistele** [openai/openai-realtime-api-beta#7]
If you are trying to use Bun Javascript to use the API Beta, please be aware that you will encounter hangs when using this package with `bun run` on the server. The reasons are related to how bun sets `globalThis.WebSocket` to use its' custom websocket implementation even on the server, but there are other issues including Authentication headers not being set properly due to Bun's implementation.

More details on it can be found here: 

https://github.com/oven-sh/bun/issues/14345
Detailed thread breaking down the issue: https://x.com/0xblacklight/status/1842031564886011947

cc @khorwood-openai I would love to find a way to get this working and considered writing a patch to use `globalThis.window` to detect browser vs. node, but this probably requires a patch on Bun's end to not hijack the `ws` package's `WebSocket` implementation, since it doesn't implement `finishRequest` for adding headers to the WebSocket request.

**Comment by elimydlarz** [comment]
I believe I'm experiencing [something similar](https://github.com/openai/openai-realtime-api-beta/issues/21) in Deno, as it also makes its `WebSocket` globally available.

**Comment by elimydlarz** [comment]
@K-Mistele is it also fixed in Deno? Just don't want to start a new issue unnecessarily.

**Comment by K-Mistele** [comment]
Bun and Deno are completely different projects. I have no idea if this problem was present in Deno at all. If so, the most productive thing to do will be do create an issue on Deno, rather than on this, since it's a runtime issue, not an issue with this repository.

**Comment by elimydlarz** [comment]
Yeah, such an issue is already open and somebody from the team mentioned they have prioritised it, I guess it will be fixed soon. Thanks.

**Comment by NoRKin** [comment]
Still not working using bun v1.1.33
